lib/Dog/dogs_list_page.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:convert';
+
+import 'package:doggo_frontend/Dog/http/dog_data.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:http/http.dart' as http;
+
+class DogsListPage extends StatefulWidget {
+  @override
+  _DogsListPageState createState() => _DogsListPageState();
+}
+
+class _DogsListPageState extends State<DogsListPage> {
+  Future<List<Dog>> _dogs;
+
+  @override
+  void initState() {
+    super.initState();
+    setState(() {
+      _dogs = _fetchDogs();
+    });
+  }
+
+  Future<List<Dog>> _fetchDogs() async {
+    final storage = FlutterSecureStorage();
+    final token = await storage.read(key: 'token');
+
+    final url = 'https://doggo-app-server.herokuapp.com/api/dogs';
+    final headers = {
+      'Content-Type': 'application/json',
+      'Accept': '*/*',
+      'Authorization': 'Bearer $token'
+    };
+
+    final response = await http.get(url, headers: headers);
+    if (response.statusCode == 200) {
+      List jsonResponse = jsonDecode(response.body);
+      return jsonResponse.map((dog) => Dog.fromJson(dog)).toList();
+    } else {
+      throw Exception('Failed to load dogs from API');
```

Comment:
Dla klienta raczej nie ma znaczenia czy nie udało się wczytać psów z API czy z innego kosmosu :P

Reply:
Exception się wyświetla jakoś użytkownikowi? Nie sprawdzałem w sumie ale myślałem że nie

Reply:
A sorry nie zauwazylem ze to jest exception, myslalem ze AlertDialogMessage